src/main/java/com/apuope/apuope_re/controllers/ResetPasswordController.java

```diff
@@ -0,0 +1,35 @@
+package com.apuope.apuope_re.controllers;
+
+import com.apuope.apuope_re.dto.ResetPasswordData;
+import com.apuope.apuope_re.dto.ResponseData;
+import com.apuope.apuope_re.jooq.tables.records.TokenRecord;
+import com.apuope.apuope_re.services.EmailService;
+import com.apuope.apuope_re.services.ResetPasswordService;
+import jakarta.mail.MessagingException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@CrossOrigin
+@RequestMapping("/api")
+public class ResetPasswordController {
+    @Autowired
+    private ResetPasswordService resetPasswordService;
+    @Autowired
+    private EmailService emailService;
+
+    @PostMapping(value = "/sendResetPasswordLink")
+    public ResponseEntity<ResponseData<String>> sendResetPasswordLink(@RequestBody String email) throws MessagingException {
+        TokenRecord response = resetPasswordService.generateEmailToken(email);
+        if (response != null) {
+            return ResponseEntity.ok(emailService.sendResetPasswordLink(response, email));
+        }
+        return ResponseEntity.ok(new ResponseData<>(false, "Error when creating email token."));
```

Comment:
This should maybe use error 500 instead of 200.

Reply:
check